src/librustdoc/html/render/mod.rs

```diff
@@ -566,17 +567,27 @@ fn document_short<'a, 'cx: 'a>(
             let (mut summary_html, has_more_content) =
                 MarkdownSummaryLine(&s, &item.links(cx)).into_string_with_has_more_content();
 
-            if has_more_content {
-                let link = format!(" <a{}>Read more</a>", assoc_href_attr(item, link, cx));
+            let link = if has_more_content {
+                let link = fmt::from_fn(|f| {
```

Comment:
NIT: This can be made more consise with [`format_args!`](https://doc.rust-lang.org/stable/std/macro.format_args.html)

Reply:
Doesn't work either...